internal/api/convert/v1alpha.go

```diff
@@ -0,0 +1,154 @@
+	out.Spec.TargetAllocator.AllocationStrategy = copy.Spec.TargetAllocator.AllocationStrategy
+	out.Spec.TargetAllocator.FilterStrategy = copy.Spec.TargetAllocator.FilterStrategy
+	out.Spec.TargetAllocator.ServiceAccount = copy.Spec.TargetAllocator.ServiceAccount
+	out.Spec.TargetAllocator.Image = copy.Spec.TargetAllocator.Image
+	out.Spec.TargetAllocator.Enabled = copy.Spec.TargetAllocator.Enabled
+	out.Spec.TargetAllocator.Affinity = copy.Spec.TargetAllocator.Affinity
+	out.Spec.TargetAllocator.PrometheusCR.Enabled = copy.Spec.TargetAllocator.PrometheusCR.Enabled
+	out.Spec.TargetAllocator.PrometheusCR.ScrapeInterval = copy.Spec.TargetAllocator.PrometheusCR.ScrapeInterval
+	out.Spec.TargetAllocator.PrometheusCR.PodMonitorSelector = copy.Spec.TargetAllocator.PrometheusCR.PodMonitorSelector
+	out.Spec.TargetAllocator.PrometheusCR.ServiceMonitorSelector = copy.Spec.TargetAllocator.PrometheusCR.ServiceMonitorSelector
+	out.Spec.TargetAllocator.SecurityContext = copy.Spec.TargetAllocator.SecurityContext
+	out.Spec.TargetAllocator.PodSecurityContext = copy.Spec.TargetAllocator.PodSecurityContext
+	out.Spec.TargetAllocator.TopologySpreadConstraints = copy.Spec.TargetAllocator.TopologySpreadConstraints
+	out.Spec.TargetAllocator.Tolerations = copy.Spec.TargetAllocator.Tolerations
+	out.Spec.TargetAllocator.Env = copy.Spec.TargetAllocator.Env
+	out.Spec.TargetAllocator.Observability = v1alpha1.ObservabilitySpec{
+		Metrics: v1alpha1.MetricsConfigSpec{
+			EnableMetrics: copy.Spec.TargetAllocator.Observability.Metrics.EnableMetrics,
+		},
+	}
+	out.Spec.TargetAllocator.PodDisruptionBudget = copy.Spec.TargetAllocator.PodDisruptionBudget
+
+	out.Spec.Mode = v1alpha2.Mode(copy.Spec.Mode)
+	out.Spec.UpgradeStrategy = v1alpha2.UpgradeStrategy(copy.Spec.UpgradeStrategy)
+	out.Spec.Ingress.Type = v1alpha2.IngressType(copy.Spec.Ingress.Type)
+	out.Spec.Ingress.Annotations = copy.Spec.Ingress.Annotations
+	out.Spec.Ingress.TLS = copy.Spec.Ingress.TLS
+	out.Spec.Ingress.IngressClassName = copy.Spec.Ingress.IngressClassName
+	out.Spec.Ingress.Route.Termination = v1alpha2.TLSRouteTerminationType(copy.Spec.Ingress.Route.Termination)
+
+	if copy.Spec.LivenessProbe != nil {
+		out.Spec.LivenessProbe = &v1alpha2.Probe{
+			InitialDelaySeconds:           copy.Spec.LivenessProbe.InitialDelaySeconds,
+			TimeoutSeconds:                copy.Spec.LivenessProbe.TimeoutSeconds,
+			PeriodSeconds:                 copy.Spec.LivenessProbe.PeriodSeconds,
+			SuccessThreshold:              copy.Spec.LivenessProbe.SuccessThreshold,
+			FailureThreshold:              copy.Spec.LivenessProbe.FailureThreshold,
+			TerminationGracePeriodSeconds: copy.Spec.LivenessProbe.TerminationGracePeriodSeconds,
+		}
+	}
+
+	out.Spec.Observability.Metrics.EnableMetrics = copy.Spec.Observability.Metrics.EnableMetrics
+
+	out.Spec.ConfigMaps = copy.Spec.ConfigMaps
+	out.Spec.DaemonSetUpdateStrategy = appsv1.DaemonSetUpdateStrategy{} // NOTE: N/A in v1alpha1
```

Comment:
This does exist in v1alpha1, it's just called [UpdateStrategy](https://github.com/jdcrouse/opentelemetry-operator/blob/main/apis/v1alpha1/opentelemetrycollector_types.go#L283)